File: check_pattern.py
import re

# Read the file
with open('content/TU-KHAINIEM/_index.md', 'r', encoding='utf-8') as f:
    content = f.read()

# Test the pattern
pattern = r'(\|\| \[.*?\]\(.*?/\) \| .*? \|)(\n\n</div>)'
match = re.search(pattern, content)

if match:
    print('✅ Pattern matched successfully!')
    print(f'Last table row: {match.group(1)}')
    print(f'After row: {match.group(2)}')
    
    # Test adding a new entry
    new_entry = '|| [Test Debug Entry](test-debug-entry/) | Đây là test... |'
    updated_content = content.replace(match.group(0), f'{match.group(1)}\n{new_entry}{match.group(2)}')
    
    # Write the updated content
    with open('content/TU-KHAINIEM/_index.md', 'w', encoding='utf-8') as f:
        f.write(updated_content)
    print('✅ Successfully added new entry to table!')
else:
    print('❌ Pattern did not match')
    print('Last 200 characters:')
    print(content[-200:])

File: check-pattern.js
const fs = require('fs');

try {
  const content = fs.readFileSync('content/TU-KHAINIEM/_index.md', 'utf8');
  
  // Test the pattern we're using
  const pattern = /(\|\| \[.*?\]\(.*?\/\) \| .*? \|)(\n\n<\/div>)/;
  const match = content.match(pattern);
  
  if (match) {
    console.log('✅ Pattern matched successfully!');
    console.log('Last table row:', match[1]);
    console.log('After row:', match[2]);
    
    // Test adding a new entry
    const newEntry = '|| [Test Entry](test-entry/) | Test content... |';
    const updatedContent = content.replace(match[0], `${match[1]}\n${newEntry}${match[2]}`);
    
    // Write the updated content
    fs.writeFileSync('content/TU-KHAINIEM/_index.md', updatedContent, 'utf8');
    console.log('✅ Successfully added new entry to table!');
  } else {
    console.log('❌ Pattern did not match');
    console.log('Last 200 characters:');
    console.log(content.substring(content.length - 200));
  }
} catch (error) {
  console.error('Error:', error.message);
}